### ISSUE TITLE
Problem with @return static

### ISSUE THREAD

**Issue by MidnightDesign** [phpstan/phpstan-strict-rules#110]
See https://phpstan.org/r/98a3d03f-add8-407f-ac45-69ac58dbba7e

```php
<?php declare(strict_types = 1);

interface FooInterface
{
    /**
     * @return static
     */
    public function deserialize();
}

final class Foo implements FooInterface
{
    public function deserialize(): self
    {
        return new self();
    }
}
```

**Expected behavior:**
No errors

**Actual behavior:**
I get the error
> Return type (Foo) of method Foo::deserialize() should be covariant with return type (static(FooInterface)) of method FooInterface::deserialize()

**Comment by ondrejmirtes** [comment]
For some reason the PHPDoc isn't inherited but should be: https://phpstan.org/r/d965bee2-214a-494f-b2ea-1fb3fe973443
